### PR TITLE
Add i386 and arm64 CI configs

### PR DESCRIPTION
**Pull request by Bodigrim**
https://cloud.drone.io/haskell/random/1

### CODE REVIEW

**Comment by lehins** [comment]
@Bodigrim Nice. I assume we need to setup an account with drone.io ? I don't have any access to repo settings, so I won't be able to do anything on that front.

**Comment by Bodigrim** [comment]
It's already set up. 

It seems there is an issue with
https://github.com/haskell/random/blob/a390fdee0dab5c661821d343d8d60d3d5ddc2ed9/src/System/Random/Internal.hs#L373-L378
`w#` is `Word64#`, so it cannot be an argument of `writeWord32Array#`, if I'm not mistaken.

**Comment by Bodigrim** [comment]
Yep, https://cloud.drone.io/haskell/random/1/2/2 fails with 
```
src/System/Random/Internal.hs:377:38: error:
    • Couldn't match expected type ‘Word#’ with actual type ‘Word64#’
    • In the third argument of ‘writeWord32Array#’, namely ‘w#’
      In the first argument of ‘io_’, namely
        ‘(writeWord32Array# mba# i32# w#)’
      In a stmt of a 'do' block: io_ (writeWord32Array# mba# i32# w#)
```

**Comment by lehins** [comment]
Beautiful, it is already useful :smile:
I'll fix it in a moment

File: .drone.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+- name: Test
+  image: buildpack-deps:focal
+  commands:
+    - export LC_ALL=C.UTF-8
+    - apt-get update -y
+    - apt-get install -y ghc cabal-install
+    - ghc --version
+    - cabal --version
+    - cabal new-update
+    - cabal new-test
```
Comment (lehins) [inline .drone.yml:14]: I think that should take care of doctest failure, right?
```suggestion
    - cabal new-test --write-ghc-environment-files=always
```

File: .drone.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  image: i386/ubuntu
+  commands:
+    - export LC_ALL=C.UTF-8
+    - apt-get update -y
+    - apt-get install -y ghc cabal-install
+    - ghc --version
+    - cabal --version
+    - cabal update
+    - cabal install --dependencies-only --enable-tests
+    - cabal test
```
Comment (lehins) [inline .drone.yml:30]: Same here:
```suggestion
    - cabal test --write-ghc-environment-files=always
```

**Comment by lehins** [comment]
> Yep, https://cloud.drone.io/haskell/random/1/2/2 fails with

Fixed and tested in https://github.com/haskell/random/pull/122

Can you also rebase on master and add a badge to readme?

Just replace the whole section with:

```
| Language | Github Actions | Drone.io | Coveralls |
|:--------:|:--------------:|:--------:|:---------:|
| ![GitHub top language](https://img.shields.io/github/languages/top/haskell/random.svg) | [![Build Status](https://github.com/haskell/random/workflows/random-CI/badge.svg)](https://github.com/haskell/random/actions) | [![Build Status](https://cloud.drone.io/api/badges/haskell/random/status.svg?ref=refs/heads/master)](https://cloud.drone.io/haskell/random/) | [![Coverage Status](https://coveralls.io/repos/github/haskell/random/badge.svg?branch=master)](https://coveralls.io/github/haskell/random?branch=master)
```
And it'll look like that:

| Language | Github Actions | Drone.io | Coveralls |
|:--------:|:--------------:|:--------:|:---------:|
| GitHub top language | [Build Status](https://github.com/haskell/random/actions) | [Build Status](https://cloud.drone.io/haskell/random/) | [Coverage Status](https://coveralls.io/github/haskell/random?branch=master)

**Comment by lehins** [comment]
Thanks @Bodigrim I took care of the rest of suggestions that I had

**Pull request merged**